File: rust/src/running_sum_of_1d_array_1480.rs

pub fn calculate_running_sum(nums: Vec<i32>) -> Vec<i32> {
    // let nums = [3, 1, 2, 10, 1];
    let mut running_sum = 0;
    let mut result: Vec<i32> = vec!();

    for num in nums {
        running_sum += num;
        // println!("{}", num);
        // println!("{}", running_sum);
        result.push(running_sum);
    }

    result
}

File: java/RunningSumOf1DArray1480.java
import java.util.ArrayList;
import java.util.List;

public class RunningSumOf1DArray1480 {
    public void calculateRunningSum() {

        int[] numbers = {3, 1, 2, 10, 1};
        int running_sum = 0;
        List<Integer> result_list = new ArrayList<>();

        // print the numbers array
        for (int num : numbers) {
            System.out.println(num);
            running_sum += num;
            result_list.add(running_sum);
        }

        System.out.println(result_list);
    }
}
